0001-typedb/02-dev/02-schema.md

```diff
@@ -0,0 +1,948 @@
+</div>
+
+The rules syntax uses `when` and `then` keywords as in `if`/`then` statement that produce virtual relations or 
+attributes when the conditions are met. Read more on how to create rules in a schema in the
+[Define rules](#define-rules) section. See the example of rules syntax below.
+
+<!-- test-ignore -->
+```typeql
+define
+
+rule rule-label:
+when {
+    ## the conditions
+} then {
+    ## the conclusion
+};
+```
+
+<div class="note">
+[Important]
+Inference can only be used in a read transaction. Learn how to use inference on the [Inferring data](06-infer.md) page.
+</div>
+
+<div class="note">
+[Note]
+The approach TypeDB uses is a [backward-chaining](https://en.wikipedia.org/wiki/Backward_chaining) execution on top 
+of [Horn-clause](https://en.wikipedia.org/wiki/Horn_clause) logic. Negation functionality follows the 
+[negation-as-failure](https://en.wikipedia.org/wiki/Negation_as_failure) model. Infinite recursion and non-termination 
+are prevented with a tabling mechanism.
+</div>
+
+## Define schema
+
+TypeQL statements must begin with the `define` keyword in order to modify a schema (e.g., create **types** or 
+**rules**). 
+
+In order to use a new TypeDB database we need to define its schema first. Use TypeDB `define` queries in a `schema` 
+session with `write` transaction to do so. The TypeQL statements of these queries must begin with the `define` 
+keyword in order to define a schema (create types or rules). 
+
+However, when running multiple statements within a TQL file, the `define` keyword only has to be included 
+once at the very beginning.
+
+<div class="note">
+[Important]
+In order to define a schema, a `schema` [session](01-connections.md#sessions) must be opened and a `write` 
+[transaction](01-connections.md#transactions) started. The changes must be **committed** or they will NOT be permanent.
+</div>
+
+There is no limitation in the order of types to define. You can define schema types in any order as long as 
+the schema as a whole is valid. TypeDB Clients validate your schema definition requests before sending them to a TypeDB 
+server. You will not be able to commit changes if the schema definition query isn’t valid.
+
+### Create a new database
+
+
+The examples below can and should be run in a separate empty database. It is recommended to create a new database 
+and to run the examples in the order they are shown. For more information on creating an empty database, see the 
+[Quickstart guide](../01-start/03-quickstart.md#create-a-database).
+
+<div class="note">
+[Important]
+Define queries are idempotent. Running the same define query a second time shall not produce any changes to the 
+database schema. Running a modified version of an already executed schema definition query can add concepts to the 
+schema but can’t modify existing ones. TypeQL schema statements do NOT replace existing type definitions but rather 
```

Comment:
This is true as-far-as types are concerned. 
But constraints such as `@key` can be removed by restating the owns without the `@key`.
e.g.:
First: `define  person owns name @key;`
Second: `define  person owns name;`

Similarly for rules, redefining a rule with the same label will overwrite it.

Reply:
1. true
2. Redefining a rule or relation?
3. That is interesting. The fact that it can be committed seems like a bug to me.

Reply:
2. I meant relation, sorry.

Reply:
Update:
3. You're right, but it should throw an exception.

Reply:
But it doesn't throw an exception/error. 
![Screenshot 2023-04-12 at 14 59 27](https://user-images.githubusercontent.com/48120135/231482031-1087c09d-90c1-4345-a297-d8a09b04fdba.png)